core/trino-main/src/main/java/io/trino/connector/WorkerDynamicCatalogManager.java

```diff
@@ -137,19 +147,26 @@ public void pruneCatalogs(Set<CatalogHandle> catalogsInUse)
             catalogCleanUpLock.unlock();
         }
 
-        // todo do this in a background thread
-        for (CatalogConnector removedCatalog : removedCatalogs) {
-            try {
-                removedCatalog.shutdown();
-            }
-            catch (Throwable e) {
-                log.error(e, "Error shutting down catalog: %s".formatted(removedCatalog));
+        var catalogsShutDown = Futures.whenAllComplete(removedCatalogs.stream()
+                .map(removedCatalog -> Futures.submit(
+                        () -> {
+                            try {
+                                removedCatalog.shutdown();
+                            }
+                            catch (Throwable e) {
+                                log.error(e, "Error shutting down catalog: %s".formatted(removedCatalog));
+                            }
+                        },
+                        executor))
+                .collect(toImmutableList()));
+
+        var ignored = catalogsShutDown.run(() ->
```

Comment:
Don't assign to a variable if it isn't going to be used.

Reply:
This was the result of some error prone check with this as the suggested fix. I'll see what I can do to remove.

Reply:
With the changing of the logging structure I just made each shutdown async and I block on none of them. Let me know if that's a little too radical or not.

Reply:
oh it still yelled even when submitting inside a forEach. Gonna try the annotation approach

Reply:
I had to do something very silly here. Let me know what you think.